scripts/lib/CIME/XML/component.py

```diff
@@ -117,12 +122,145 @@ def _get_value_match(self, node, attributes=None, exact_match=False):
+                desc = node.text
+        # cpl and esp components may not have a description
+        if comp_class not in ['cpl','esp']:
+            expect(len(desc) > 0,
+                   "No description found for comp_class {} matching compsetname {} in file {}"\
+                       .format(comp_class,compsetname, self.filename))
+        return desc
+
+    def _get_description_match(self, compsetname, reqset, fullset, modifier_mode):
+        """
+
+        >>> obj = Component('testingonly', 'ATM')
+        >>> obj._get_description_match("1850_DATM%CRU_FRED",set(["DATM"]), set(["DATM","CRU","HSI"]), "*")
+        True
+        >>> obj._get_description_match("1850_DATM%FRED_Barn",set(["DATM"]), set(["DATM","CRU","HSI"]), "*")
+        False
+        >>> obj._get_description_match("1850_DATM_Barn",set(["DATM"]), set(["DATM","CRU","HSI"]), "?")
+        True
+        >>> obj._get_description_match("1850_DATM_Barn",set(["DATM"]), set(["DATM","CRU","HSI"]), "1")
+        Traceback (most recent call last):
+        ...
+        SystemExit: ERROR: Expected exactly one modifer found 0
+        >>> obj._get_description_match("1850_DATM%CRU%HSI_Barn",set(["DATM"]), set(["DATM","CRU","HSI"]), "1")
+        Traceback (most recent call last):
+        ...
+        SystemExit: ERROR: Expected exactly one modifer found 2
+        >>> obj._get_description_match("1850_CAM50%WCCM%RCO2_Barn",set(["CAM50", "WCCM"]), set(["CAM50","WCCM","RCO2"]), "*")
+        True
+
+        # The following is not allowed because the required WCCM field is missing
+        >>> obj._get_description_match("1850_CAM50%RCO2_Barn",set(["CAM50", "WCCM"]), set(["CAM50","WCCM","RCO2"]), "*")
+        False
+        >>> obj._get_description_match("1850_CAM50_Barn",set(["CAM50", "WCCM"]), set(["CAM50","WCCM","RCO2"]), "+")
+        False
+        >>> obj._get_description_match("1850_CAM50%WCCM_Barn",set(["CAM50", "WCCM"]), set(["CAM50","WCCM","RCO2"]), "+")
+        True
+        """
+        match = False
+        comparts = compsetname.split('_')
+        for comp in comparts:
+            complist = comp.split('%')
+            cset = set(complist)
+            if cset == reqset or (cset > reqset and cset <= fullset):
+                if modifier_mode == '1':
+                    expect(len(complist) == 2,
+                           "Expected exactly one modifer found {}".format(len(complist)-1))
+                elif modifier_mode == '+':
+                    expect(len(complist) >= 2,
+                           "Expected one or more modifers found {}".format(len(complist)-1))
+                elif modifier_mode == '?':
+                    expect(len(complist) <= 2,
+                           "Expected 0 or one modifers found {}".format(len(complist)-1))
+                expect(not match,"Found multiple matches in file {} for {}".format(self.filename,comp))
+                match = True
+                # found a match
```

Comment:
In this one we want to pass all the way through to make sure that there is only one match and throw an error if there is more than one.